rootfs/templates/wrapper-body.sh

```
@@ -0,0 +1,591 @@
+	trap run_exit_hooks EXIT
+
+	if [ "$ONE_SHELL" != "true" ]; then
+		CONTAINER_ID=$(docker ps --filter name="^/${DOCKER_NAME}\$" --format '{{ .ID }}')
+		if [ -n "$CONTAINER_ID" ]; then
+			echo "# Starting shell in already running ${DOCKER_NAME} container ($CONTAINER_ID)"
+			if [ $# -eq 0 ]; then
+				set -- "/bin/bash" "-l" "$@"
+			fi
+			# We set unusual detach keys because (a) the default first char is ctrl-p, which is used for command history,
+			# and (b) if you detach from the shell, there is no way to reattach to it, so we want to effectively disable detach.
+			docker exec -it --detach-keys "ctrl-^,ctrl-[,ctrl-@" --env GEODESIC_HOST_CWD="${GEODESIC_HOST_CWD}" "${DOCKER_NAME}" $*
```

Shellcheck (0.10.0)

[warning] Use "$@" (with quotes) to prevent whitespace problems.

(SC2048)